runtime/runtime-params-estimator/src/gas_cost.rs

```diff
@@ -422,16 +378,18 @@ impl ops::Add for GasCost {
     type Output = GasCost;
 
     fn add(mut self, rhs: GasCost) -> Self::Output {
-        assert_eq!(self.metric, rhs.metric);
         self.combine_uncertain(&rhs);
-        GasCost {
-            time_ns: self.time_ns + rhs.time_ns,
-            instructions: self.instructions + rhs.instructions,
-            io_r_bytes: self.io_r_bytes + rhs.io_r_bytes,
-            io_w_bytes: self.io_w_bytes + rhs.io_w_bytes,
-            metric: self.metric,
-            uncertain: self.uncertain,
-        }
+        let qemu_measurement = match (self.qemu_measurement, rhs.qemu_measurement) {
+            (None, None) => None,
+            (Some(lhs), Some(rhs)) => Some(lhs + rhs),
+            (single_value, None) | (None, single_value) => single_value,
+        };
```

Comment:
Do we need `zip_with(self.qemu_measurement, rhs.qemu_measurement, Add::add)` helper hof?

Reply:
I don't know, do we? It feels like the code wouldn't necessarily become easier to read. 
I tend towards simple and blunt solutions over elegance in this case.